### PR TITLE
Minor additions to regression analyses

### PR DESCRIPTION
**Pull request by TomiSzucs**
Added unit test for White's test of heteroscedasticity and added decision for hypothesis tests in case of heteroscedasticity.

### CODE REVIEW

File: cogstat/cogstat_hyp_test.py [hunk 1]
```diff
@@ -431,14 +435,26 @@ def variable_pair_hyp_test(data, x, y, meas_lev, normality=None):
...
-
             r, p = stats.spearmanr(data[x], data[y])
             population_result += _("Spearman's rank-order correlation") + \
                                  ': <i>r<sub>s</sub></i>(%d) = %0.*f, %s' % \
                                  (df, non_data_dim_precision, r, print_p(p))
 
-        else:  # TODO or normality couldn't be calculated
-            population_result += '<decision>'+_('Interval variables.') + ' ' + _('Normality violated.') + ' >> ' + \
+        elif normality is None or homoscedasticity is None:
+            # TODO warning instead of omitting hypothesis tests?
```
Comment (krajcsi) [inline cogstat/cogstat_hyp_test.py:444]: I agree that, at the moment, a warning would be better, and hypothesis tests could be calculated.

File: cogstat/cogstat_hyp_test.py [hunk 2]
```diff
@@ -431,14 +435,26 @@ def variable_pair_hyp_test(data, x, y, meas_lev, normality=None):
...
+            population_result += '<decision>'+_('Interval variables.') + ' ' \
+                                 + _('Assumptions of hypothesis tests could not be tested.') + ' >> ' \
+                                 + _("Hypothesis tests not run.") + '\n</decision>'
+
+        else:
+            violations = ""
```
Comment (krajcsi) [inline cogstat/cogstat_hyp_test.py:450]:
```suggestion
            violations = ''
```

**Review by krajcsi — CHANGES_REQUESTED** [review]
Looks good to me. Only a minor modification should be considered.

File: cogstat/cogstat_hyp_test.py (hunk 1 above)
Comment (TomiSzucs) [reply cogstat/cogstat_hyp_test.py:444]: Modified to warning instead of decision.

**Pull request merged**